lib/basic-auth-request.js

```diff
@@ -2,12 +2,22 @@
 
 const request = require('request');
 
-const buildError = (body) => {
-  const err = new Error(body.message);
-  err.statusCode = body.code;
+/**
+ * Builds a custom error based on the `error` provided by request module.
+ * @param {Object} requestError The request error object from request module.
+ * @returns {Object} The custom error.
+ */
+const buildError = (requestError) => {
+  const err = new Error(requestError.message);
+  err.statusCode = requestError.code;
   return err;
 };
 
+/**
+ * Builds a custom request object and makes a http request to retrieve an access token.
+ * @param {Object} settings The settings object which contains username and password.
+ * @returns {Promise} Promise object to resolve with the new access token.
+ */
```

Comment:
I'm not sure we need to do anything in this file since we are not exposing any of these functions publicly

Reply:
Makes sense, thanks for the review, I'm going to remove the docs. :+1: